sockeye/training.py

```diff
@@ -281,6 +282,10 @@ def _fit(self,
         """
         metric_train = self._create_eval_metric(metrics)
         metric_val = self._create_eval_metric(metrics)
+        # Track loss for optimizer if needed
+        if isinstance(self.module._curr_module._optimizer, SockeyeOptimizer):
+            # Create metric corresponding to training loss
+            metric_loss = mx.metric.create(self.config.config_loss.type, loss_config=self.config.config_loss)
```

Comment:
would it make sense to create a constant (with better naming ;)) `OPTIMIZER_LOSSES_WITH_METRIC = [C.CROSS_ENTROPY, C.SMOOTHED_CROSS_ENTROPY]` in order to check that `self.config.config_loss.type in OPTIMIZER_LOSSES_WITH_METRIC` to get a better error message if we ever add new losses, which don't have a corresponding metric. Or is the assumption that every loss will have a corresponding metric? If so this should be documented in `class Loss`. Or we could make it explicit by:
```
class Loss(ABC):
   ...
   @abstract_method
   def create_metric(self):
       pass
```

Reply:
Good idea.  That looks cleaner.